### PR TITLE
Ava

### PR DESCRIPTION
**Pull request by sveisvei**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,14 +39,13 @@
...
+    "test": "PORT=4243 ava **/**test.js",
     "test:docker": "./scripts/docker-postgres.sh",
-    "test:watch": "mocha --watch test test/*",
+    "test:watch": "npm run test -- --watch",
     "test:pg-virtualenv": "pg_virtualenv npm run test:pg-virtualenv-chai",
     "test:pg-virtualenv-chain": "export TEST_DATABASE_URL=postgres://$PGUSER:$PGPASSWORD@localhost:$PGPORT/postgres ; npm run db-migrate-testdb && npm test",
-    "test:coverage": "istanbul cover ./node_modules/mocha/bin/_mocha test --report lcovonly -- -R spec --recursive",
-    "test:coverage-report": "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js && rm -rf ./coverage"
+    "coverage": "nyc npm run test",
+    "coverage-report": "nyc report --reporter=text-lcov | coveralls && rm -rf ./coverage"
```
Comment (ivarconr) [inline package.json]: remember to update `.travis.yaml` also!

File: test/unit/routes/mocks/fake-feature-toggle-store.js [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
 
-const _features = [];
 
-module.exports = {
-    getFeatures: () => Promise.resolve(_features),
-    addFeature: (feature) => _features.push(feature),
-    reset: () => {
-        _features.lengyh = 0;
-    },
+module.exports =  () => {
```
Comment (ivarconr) [inline test/unit/routes/mocks/fake-feature-toggle-store.js]: yes! I was thinking of making these in to a closure.

File: test/e2e/util/test-helper.js [hunk 3]
```diff
@@ -2,28 +2,39 @@
...
+    const db = createDb({ databaseUri: options.databaseUri, minPool: 0, maxPool: 0 });
 
-const createApp = db.raw(`CREATE SCHEMA IF NOT EXISTS ${options.databaseSchema}`)
-    .then(() => migrator(options.databaseUri, options.databaseSchema))
-    .then(() => {
-        const stores = createStores(options);
-        const app = require('../../../app')({stores});
-        return { stores, request: supertest(app) };
-    });
+    return db.raw(`CREATE SCHEMA IF NOT EXISTS ${options.databaseSchema}`)
```
Comment (ivarconr) [inline test/e2e/util/test-helper.js]: maybe drop schema first if it exists, then create?

Comment (ivarconr) [inline test/e2e/util/test-helper.js]: `DROP SCHEMA IF EXISTS ${options.databaseSchema} CASCADE; CREATE SCHEMA ${options.databaseSchema}`

**Comment by ivarconr** [comment]
Nice job! 

Running in each test in separate schemas at least test that our migrations works!

File: test/e2e/util/test-helper.js (hunk 3 above)
Comment (sveisvei) [inline test/e2e/util/test-helper.js]: if we drop schema, test reruns will be slower, but maybe we then could remove the cleanup stuff?

btw know how to get away logging from the migration runner?

Comment (sveisvei) [inline test/e2e/util/test-helper.js]: I left the schema reset out, lets bring it back in a other PR maybe?

**Comment by ivarconr** [comment]
Jeah, there is a hack for that:

require('db-migrate-shared').log.silence(true);

**Comment by sveisvei** [comment]
Nice. added the log hack in https://github.com/Unleash/unleash/commit/bb4cf30d2228851cc2eb90d9f254f5e4d0ce81a4

**Pull request merged**